src/coreclr/jit/importer.cpp

```diff
@@ -21115,7 +21115,17 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
                                 //
                                 if ((Target::g_tgtArgOrder == Target::ARG_ORDER_R2L) || (call->gtCallArgs == nullptr))
                                 {
-                                    call->gtCallArgs = gtPrependNewCallArg(methodTableArg, call->gtCallArgs);
+                                    // If there's a ret buf, the context is the second arg.
+                                    //
+                                    if (call->HasRetBufArg())
+                                    {
+                                        GenTreeCall::Use* const beforeArg = call->gtCallArgs;
+                                        beforeArg->SetNext(gtNewCallArgs(methodTableArg));
```

Comment:
would `gtInsertNewCallArgAfter(methodTableArg, call->gtCallArgs)`  work here? If I understand correctly 'gtCallArgs` points to `retBufArg` and we need to insert `methodTableArg` after it.

Reply:
Yes, that would work. You prefer it that way?

Reply:
yes, it would be similar to 'else' then I think.